lib/sam/common/pmc.c

```diff
@@ -44,20 +44,29 @@ void pmc_plla_config(uint8_t mul, uint8_t div)
 
 void pmc_peripheral_clock_enable(uint8_t pid)
 {
+#if defined(SAM3N) || defined(SAM3U)
+	//SAM3N and SAM3U only have one Peripheral Clock Enable Register
+	PMC_PCER = 1 << pid;
+#else
```

Comment:
Is this actually necessary?  I mean, sure, you save a tiny tiny amount of flash, but pid will always be lower than 32 on 3n and 3u anyway right?  (Or are the defines missing?  If so, it might be better to check for #if defined(PMC_PCER1) rather than each family